### DISCUSSION TITLE
Timezone UTC not working without -Duser.timezone=UTC

### DISCUSSION THREAD

**Discussion by Paul6552** [quarkusio/quarkus#39923] [Q&A]
Unfortunately, since upgrading Quarkus to the latest LTS 3.8.3 version, I've been having a hard time with timezones.

My old code which worked with the old LTS version:
```
    public App() {
        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
    }
    private void registerFilters(HashSet<Class<?>> set) {
        set.add(JacksonJavaTimeMappingProvider.class);
    }
@Provider
public class JacksonJavaTimeMappingProvider implements ContextResolver<ObjectMapper> {

    private final ObjectMapper mapper;

    public JacksonJavaTimeMappingProvider() {
        mapper = new ObjectMapper();

        mapper.registerModule(new JavaTimeModule());
        mapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
        mapper.setTimeZone(TimeZone.getTimeZone(ZoneId.systemDefault()));
    }

    @Override
    public ObjectMapper getContext(Class<?> type) {
        return mapper;
    }
}
```

Since upgrading to the latest version, I have had the problem that Hibernate saves all dates in the database with European time.

I was able to fix this again with:
`quarkus.hibernate-orm.jdbc.timezone=UTC`

However, when I load the dates from the database, my application converts the date back to the "European time zone". I thought I could solve this with the following Quarkus Property:
`quarkus.hibernate-orm.default-timezone=UTC`

In the end, whether I delete the `Jacksonmapper` or the programmatic statement (
`TimeZone.setDefault(TimeZone.getTimeZone("UTC"));`) in the application, I can't get a UTC time.

The only thing that worked was setting the JAVA_OPTS:
`-Duser.timezone=UTC`

That was able to fix my application. The problem is, this wasn't necessary before. I feel a bit uncomfortable using java_opts and would like to solve it programmatically in the app again.

Did I miss a Quarkus property or...?
Thank you very much

**Comment by gsmet** [comment]
From which version did you upgrade?

I think it would be interesting to have a small Maven reproducer of what was working before and what is now failing. Ideally with a small test that works with the previous version and fails with the new one.

**Reply by Paul6552** [reply]
I migrated from 3.2.11.
